### DISCUSSION TITLE
When will CLANG support come?

### DISCUSSION THREAD

**Discussion by meftunca** [jfalcou/eve#1521] [Done & Done!]
I've been waiting for the clang-related regulation to be distributed with the new version for a long time. Please share this edit now, even with the patch.

**Comment by jfalcou** [comment]
Which clang version ? We do support clang already for a long time.

**Reply by meftunca** [reply]
Clang Info:
1. Homebrew clang version 15.0.6
2. Target: x86_64-apple-darwin22.1.0
3. Thread model: posix
4. InstalledDir: /usr/local/opt/llvm/bin

Conan Error: 
> jfalcou-eve/v2022.09.1: Invalid ID: EVE does not support apple Clang due to an incomplete libcpp.

**Reply by jfalcou** [reply]
That's the Conan package recipe indeed.
The machine used by the CIC doesn't have the correct version of clang so I can't make it run there.
Did you try with a regular installation ?

**Reply by meftunca** [reply]
I am using bazel in my project. Since I got a similar error about clang when I tried to install it externally, I decided to stay in conan and wait for the fix to come. I will install it locally again in the evening and share the error with you.

**Reply by DenisYaroshevskiy** [reply]
Can you try to git clone instead? I know i's annoying but we Apple Clang is a bit tricky. Latest versions should mostly work (the only thing on my M1 didn't work was runtime arch selection. Some x86 at least built succesfully).

**Reply by meftunca** [reply]
Unfortunately, I could not reach a successful result because I got index errors in the installation with bazel. If there is an installation example for Bazel, I can review it.

**Reply by meftunca** [reply]
./internal/eve/module/core.hpp:94:10: fatal error: 'eve/module/core/constant/core.hpp' file not found
#include <eve/module/core/constant/core.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

**Reply by DenisYaroshevskiy** [reply]
Here is that file

https://github.com/jfalcou/eve/blob/e015b18183df70ff3066e3b7192812d6cfae88d9/include/eve/module/core/constant/core.hpp

I don't know anything about bazel. Can you tell it "all files in include"?

**Reply by meftunca** [reply]
Yes I have included all the files in it
```
load("@rules_cc//cc:defs.bzl", "cc_library")

cc_library(
    name = "eve",
    hdrs = glob(["**/*.hpp"]),
    visibility = ["//visibility:public"],

)
```

**Reply by DenisYaroshevskiy** [reply]
I don't know how that works, but what counts as a root?
We assume -I eve/include

**Reply by meftunca** [reply]
Finally got it done :). Finally it's my turn to write the Json Parser. Now I can learn your library. :)

I would like to briefly talk about why I use your project. You have the most developer friendly library for simd I've ever seen. I am writing a new generation database model with C++. I finished the simple read/write operations and now I want to go to the 2nd stage by solving the performance problems. The biggest problem that anyone who develops a database can admit is accessing/filtering/sorting stacks. I will use your library to solve this problem via simd.

**Reply by DenisYaroshevskiy** [reply]
Fantastic. You can look at collect indexes example and remove algorithm to do filtering and such. Good luck
